Formula/c/charmcraft.rb

```diff
@@ -17,20 +17,24 @@ class Charmcraft < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux:   "7d6ce9798ec325abab4604c95e8754ce5872e2563b22cd4e43a4560c4cbae358"
   end
 
-  depends_on "rust" => :build # for rpds-py
-  depends_on "cffi"
-  depends_on "keyring"
+  depends_on "rust" => :build
   depends_on "libsodium"
-  depends_on "pycparser"
+  depends_on "libyaml" # for rpds-py
```

Comment:
I'm not crazy about these comments personally, and have been removing them in my PR's.. Not a blocker though if you'd prefer to keep it. I will note that if you want to keep this it should be moved up to `rust` since that's what it actually applies to. `brew style --fix` doesn't respect the placement of comments unfortunately.

Reply:
Thanks for specifically calling out that the comment moved! I originally figured that the comment is valuable as a future hint to someone. But, I'd been in a habit of just running `brew style --fix` sometimes and not taking into account that that can't handle moving comments around. I'll just nerf the comment for this one.

Reply:
Yeah, would be nice to find a way to preserve the location of comments, because this happens all the time. CC @Homebrew/brew for possible ideas.

Reply:
> Yeah, would be nice to find a way to preserve the location of comments, because this happens all the time.

This is something @issyl0 or @razvanazamfirei dealt with on a recent PR iirc?

Reply:
https://github.com/Homebrew/brew/blob/f138059de1efc4e2758d03cf0fa9e5b363155058/Library/Homebrew/rubocops/cask/uninstall_methods_order.rb#L40-L55
https://github.com/Homebrew/brew/blob/f138059de1efc4e2758d03cf0fa9e5b363155058/Library/Homebrew/rubocops/cask/array_alphabetization.rb#L37-L58

Would need to look and see the other contexts comments are used and maybe write something more generic.